.circleci/config.yml

```diff
@@ -38,17 +39,44 @@ jobs:
             - build-info.json
             - dist
             - assets/stylesheets
+  vulnerability_scan:
+    executor: hmpps/node
+    parameters:
+      monitor:
+        type: boolean
+        default: false
+    steps:
+      - checkout
+      - npm install -q
+      - snyk/scan:
+          project: '${CIRCLE_PROJECT_REPONAME}'
+          monitor-on-build: << parameters.monitor >>
+          organization: "digital-probation-services"
+          severity-threshold: "high" # note: this does not affect snyk 'monitor' commands
+          fail-on-issues: true
 
 workflows:
   version: 2
-  build-test-and-deploy:
+  build_test_and_deploy:
     jobs:
       - build:
         filters:
           tags:
             ignore: /.*/
 #      - hmpps/helm_lint:
 #          name: helm_lint
+      - vulnerability_scan:
+          name: vulnerability_scan_and_monitor
+          monitor: true
+          filters:
+            branches:
+              only:
+                - main
+      - vulnerability_scan:
+          filters:
+            branches:
+              ignore:
+                - main
```

Comment:
Sorry, I'm not an expert on circle ci, so I'm a bit confused about these 2 sections - does this not undo the running of `vulnerability_scan`'s`only` above i.e. not run on `main`?

Reply:
the filters are mutually exclusive; one runs only on main, the other on everything except main. the idea is just to run the job with different params for PR builds.

Reply:
Thanks for clarifying!